cumulus/pallets/dmp-queue/src/lib.rs

```diff
@@ -1,915 +1,239 @@
 // Copyright (C) Parity Technologies (UK) Ltd.
-// This file is part of Cumulus.
-
-// Cumulus is free software: you can redistribute it and/or modify
-// it under the terms of the GNU General Public License as published by
-// the Free Software Foundation, either version 3 of the License, or
-// (at your option) any later version.
-
-// Cumulus is distributed in the hope that it will be useful,
-// but WITHOUT ANY WARRANTY; without even the implied warranty of
-// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-// GNU General Public License for more details.
-
-// You should have received a copy of the GNU General Public License
-// along with Cumulus.  If not, see <http://www.gnu.org/licenses/>.
-
-//! Pallet implementing a message queue for downward messages from the relay-chain.
-//! Executes downward messages if there is enough weight available and schedules the rest for later
-//! execution (by `on_idle` or another `handle_dmp_messages` call). Individual overweight messages
-//! are scheduled into a separate queue that is only serviced by explicit extrinsic calls.
+// SPDX-License-Identifier: Apache-2.0
```

Comment:
I know, we may probably ignore it - just wanted to mention that this migration (unless I miss something) violates XCM " XCM messages are guaranteed to be delivered and interpreted accurately, in order" rule - because while migration is active, the parachain keep receiving new DMP messages and they're inserted in the MQ pallet. So in the MQ could look like: `[<bunch-of-new-messages>, <bunch-of-old-messages>, <bunch-of-new-messages>]`, so they won't be "interpreted in order". But as I said, imo this could be probably ignored because this migration is a one-time thing. Unless we know that it may violate something - I wonder if we have some code that is relying on this "in order" requirement

Reply:
Yep the DMP queue migration violates this indeed. I can just try to add a `.rev()` to the message iteration, then at least the non-overweight messages should stay in order. Although they can still be interlaced with other messages.

Reply:
Don't know if we need it :) I mean - does it matter to keep order of old messages if they are mixed with new ones? So don't make any changes just because of my comment :)

Reply:
Yea it would not really improve the situation... maybe i will leave it as it is...